packages/common/src/while-in-flight.js

```diff
@@ -0,0 +1,34 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+/* eslint no-invalid-this: [0] */
+
+import {wrap} from 'lodash';
+import tap from './tap';
+
+/**
+ * While the promise returned by the decorated is unfullfilled, sets, the
+ * specified boolean on the target class to `true`
+ * @param {string} param
+ * @returns {Function}
+ */
+export default function whileInFlight(param) {
```

Comment:
This is pretty interesting. Seems like a good starting point and I'm intrigued to see where else we end up using it.